tools/helper.psm1

```diff
@@ -81,6 +82,92 @@ function Install-Dotnet {
     }
 }
 
+function New-gRPCAutoGenCode
+{
+    [CmdletBinding()]
+    param(
+        [switch] $Force
+    )
+
+    if ($Force -or -not (Test-Path "$RepoRoot/src/Messaging/FunctionRpc.cs") -or
+                   -not (Test-Path "$RepoRoot/src/Messaging/FunctionRpcGrpc.cs"))
+    {
+        Write-Log "Generate the CSharp code for gRPC communication from protobuf"
+
+        Resolve-ProtoBufToolPath
+
+        $outputDir = "$RepoRoot/src/Messaging"
+        Remove-Item "$outputDir/FunctionRpc*.cs" -Force -ErrorAction SilentlyContinue
+
+        & $Script:protoc_Path $Script:protobuf_file_Path --csharp_out $outputDir `
+                                                         --grpc_out=$outputDir `
+                                                         --plugin=protoc-gen-grpc=$Script:grpc_csharp_plugin_Path `
+                                                         --proto_path=$Script:protobuf_dir_Path `
+                                                         --proto_path=$Script:google_protobuf_tools_Path
+        if ($LASTEXITCODE -ne 0) {
+            throw "Failed to generate the CSharp code for gRPC communication."
+        }
+    }
+}
+
+function Resolve-ProtoBufToolPath
+{
+    if (-not $Script:protoc_Path) {
+        Write-Log "Resolve the protobuf tools for auto-generating code"
```

Comment:
it can probably be addressed in another PR but is there a reason we have `Write-Log`? Couldn't we just use `Write-Host` and `Write-Warning` etc?

Reply:
I don't want to specify `-ForegroundColor Green` for all the intentional progress logging. `Write-Log` abstracts that flag.